crates/pathfinder/src/config/cli.rs

```diff
@@ -50,109 +50,193 @@ where
-    clap::App::new("Pathfinder")
+    clap::Command::new("Pathfinder")
         .version(version)
         .about("A StarkNet node implemented by Equilibrium. Submit bug reports and issues at https://github.com/eqlabs/pathfinder.")
         .arg(
-            Arg::with_name(CONFIG_KEY)
-                .short("c")
+            Arg::new(CONFIG_KEY)
+                .short('c')
                 .long(CONFIG_KEY)
                 .help("Path to the configuration file.")
                 .value_name("FILE")
                 .takes_value(true),
         )
         .arg(
-            Arg::with_name(ETH_USER_KEY)
+            Arg::new(ETH_USER_KEY)
                 .long(ETH_USER_KEY)
                 .help("Ethereum API user")
                 .takes_value(true)
+                .env("PATHFINDER_ETHEREUM_API_USERNAME")
                 .long_help("The optional user to use for the Ethereum API"),
         )
         .arg(
-            Arg::with_name(ETH_PASS_KEY)
+            Arg::new(ETH_PASS_KEY)
                 .long(ETH_PASS_KEY)
                 .help("Ethereum API password")
                 .takes_value(true)
+                .env("PATHFINDER_ETHEREUM_API_PASSWORD")
                 .long_help("The optional password to use for the Ethereum API"),
         )
         .arg(
-            Arg::with_name(ETH_URL_KEY)
+            Arg::new(ETH_URL_KEY)
                 .long(ETH_URL_KEY)
                 .help("Ethereum API endpoint")
                 .takes_value(true)
                 .value_name("HTTP(s) URL")
+                .env("PATHFINDER_ETHEREUM_API_URL")
                 .long_help(r"This should point to the HTTP RPC endpoint of your Ethereum entry-point, typically a local Ethereum client or a hosted gateway service such as Infura or Cloudflare.
 Examples:
     infura: https://goerli.infura.io/v3/<PROJECT_ID>
     geth:   https://localhost:8545"))
         .arg(
-            Arg::with_name(HTTP_RPC_ADDR_KEY)
+            Arg::new(HTTP_RPC_ADDR_KEY)
                 .long(HTTP_RPC_ADDR_KEY)
-                .help(&HTTP_RPC_HELP)
+                .help(HTTP_RPC_HELP.as_ref())
                 .takes_value(true)
                 .value_name("IP:PORT")
+                .env("PATHFINDER_HTTP_RPC_ADDRESS")
         )
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
+    use std::env;
+    use std::sync::Mutex;
+
+    lazy_static::lazy_static! {
+        // prevents running tests in parallel, since these depend on
+        // process-global environment variables
+        static ref ENV_VAR_MUTEX: Mutex<()> = Mutex::new(());
+    }
+
+    fn clear_environment() {
+        env::remove_var("PATHFINDER_ETHEREUM_API_USERNAME");
+        env::remove_var("PATHFINDER_ETHEREUM_API_PASSWORD");
+        env::remove_var("PATHFINDER_ETHEREUM_API_URL");
+        env::remove_var("PATHFINDER_HTTP_RPC_ADDRESS");
+    }
 
     #[test]
     fn ethereum_url_long() {
+        let _env_guard = ENV_VAR_MUTEX.lock().unwrap();
```

Comment:
```suggestion
        let _env_guard = ENV_VAR_MUTEX.lock().unwrap_or_else(|e| e.into_inner());
```

I think this would be preferable, otherwise panic in one will panic others which follow it.

Reply:
Thanks, this is a good idea, thanks!